binaries/cuprated/src/p2p/request_handler.rs

```diff
@@ -1 +1,432 @@
+async fn new_fluffy_block<A: NetZoneAddress>(
+    peer_information: PeerInformation<A>,
+    request: NewFluffyBlock,
+    mut blockchain_read_handle: BlockchainReadHandle,
+    mut txpool_read_handle: TxpoolReadHandle,
+) -> anyhow::Result<ProtocolResponse> {
+    // TODO: check context service here and ignore the block?
+    let current_blockchain_height = request.current_blockchain_height;
+
+    peer_information
+        .core_sync_data
+        .lock()
+        .unwrap()
+        .current_height = current_blockchain_height;
+
+    let (block, txs) = rayon_spawn_async(move || -> Result<_, anyhow::Error> {
+        let block = Block::read(&mut request.b.block.as_ref())?;
+
+        let tx_blobs = request
+            .b
+            .txs
+            .take_normal()
+            .ok_or(anyhow::anyhow!("Peer sent pruned txs in fluffy block"))?;
+
+        let txs = tx_blobs
+            .into_iter()
+            .map(|tx_blob| {
+                if tx_blob.len() > MAX_TRANSACTION_BLOB_SIZE {
+                    anyhow::bail!("Peer sent a transaction over the size limit.");
+                }
+
+                let tx = Transaction::read(&mut tx_blob.as_ref())?;
+
+                Ok((tx.hash(), tx))
+            })
+            .collect::<Result<_, anyhow::Error>>()?;
+
+        // The backing `Bytes` will be deallocated when this closure returns.
+
+        Ok((block, txs))
+    })
+    .await?;
+
+    let res = blockchain_interface::handle_incoming_block(
+        block,
+        txs,
+        &mut blockchain_read_handle,
+        &mut txpool_read_handle,
+    )
+    .await;
+
+    match res {
+        Ok(_) => Ok(ProtocolResponse::NA),
+        Err(IncomingBlockError::UnknownTransactions(block_hash, missing_tx_indices)) => Ok(
+            ProtocolResponse::FluffyMissingTransactionsRequest(FluffyMissingTransactionsRequest {
+                block_hash: block_hash.into(),
+                current_blockchain_height,
+                missing_tx_indices: missing_tx_indices.into_iter().map(usize_to_u64).collect(),
+            }),
+        ),
+        Err(IncomingBlockError::Orphan) => {
+            // Block's parent was unknown, could be syncing?
+            Ok(ProtocolResponse::NA)
+        }
+        Err(e) => Err(e.into()),
+    }
+}
+
+/// [`ProtocolRequest::NewTransactions`]
+async fn new_transactions<A>(
+    peer_information: PeerInformation<A>,
+    request: NewTransactions,
+    mut blockchain_context_service: BlockChainContextService,
+    mut incoming_tx_handler: IncomingTxHandler,
+) -> anyhow::Result<ProtocolResponse>
+where
+    A: NetZoneAddress,
+    InternalPeerID<A>: Into<CrossNetworkInternalPeerId>,
+{
+    let BlockChainContextResponse::Context(context) = blockchain_context_service
+        .ready()
+        .await
+        .expect(PANIC_CRITICAL_SERVICE_ERROR)
+        .call(BlockChainContextRequest::Context)
+        .await
+        .expect(PANIC_CRITICAL_SERVICE_ERROR)
+    else {
+        unreachable!()
+    };
```

Comment:
From https://github.com/Cuprate/cuprate/pull/308#pullrequestreview-2463412554:

> [binaries/cuprated/src/rpc/request/*.rs](https://github.com/Cuprate/cuprate/tree/main/binaries/cuprated/src/rpc/request) which are fn versions of our tower::Service's Request -> Response to reduce noise

Since `cuprated` will be using `Service`s like this in general, should I move this folder? Maybe `cuprated/src/service/*.rs`? I think this will reduce noise significantly through `cuprated`, example here would be:
```rust
let context = blockchain_context::context(&mut blockchain_context_service)
    .await
    .expect(PANIC_CRITICAL_SERVICE_ERROR)
    .unchecked_blockchain_context();
```

Reply:
I have plans to rework this service to make getting the context not require a `Service` request, so for now I don't think so.